### ISSUE TITLE
Files can't be opened

### ISSUE THREAD

**Issue by nicolascolla** [nicolascolla/simple-reader#2]
### Expected behavior

App launches, "open a book" button is pressed, file is selected, file is opened.

### Current behavior

App launches, "open a book" button is pressed, file is selected, nothing happens, "open a book" button disappears.

### Affected devices

- **BQ Aquaris M10** (RC) - LionelB
- **OnePlus 3** (RC) - Moem
- **OnePlus 5** (RC) - LionelB
- **Xperia X** (RC) - LionelB
- **Xperia X** (devel) - Danfro

### Working devices

- **BQ E5** (Devel image 17-11-21) - Tobiyo Kuujikai
- **OnePlus 1** (RC) - Moem
- **OnePlus 6t**  (OTA-19) - Steven Bersson
- **Volla phone** (OTA-19) - @gbdomubpkm
- **Xiaomi Redmi Note 9** (RC) - Nicolás Colla
- **Xiaomi Redmi 9** (RC) - Tobiyo Kuujikai

### Observations

This bug can be reproduced with [ePubViewer](https://github.com/gsantner/ePubViewer), while [the web version ](https://pgaskin.net/ePubViewer/) works without issues inside Morph on all devices.

Log file (`~/.cache/upstart/application-click-simplereader.collaproductions_simplereader_1.0.2.log`) of a working device (Redmi Note 9):

```
Creating a QMirClientScreen now
Started as a local application container.
qml: Loading webview on /home/phablet/.local/share/simplereader.collaproductions
Sandboxing disabled by user.
[8880:8931:1122/231426.618390:ERROR:file_path_watcher_linux.cc(71)] Failed to read /proc/sys/fs/inotify/max_user_watches
qq= WebViewImplOxide_QMLTYPE_107(0x3afcba60) 0x3b10ff30
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onContextMenuIntercepted"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onInsertionHandleTapped"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onStatusChanged"
qml: Start browsing
QDir::mkpath: Empty or null file name
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:145:9: Unable to assign [undefined] to QObject*
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.2154\" (uid=32011 pid=8880 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.Operations\" member=\"list\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=5640 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.2154\" (uid=32011 pid=8880 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"running\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=5640 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.2154\" (uid=32011 pid=8880 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"last_error\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=5640 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.2154\" (uid=32011 pid=8880 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"result\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=5640 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
[8880:8941:1122/231426.811589:ERROR:zygote_host_impl_linux.cc(259)] Failed to adjust OOM score of renderer with pid 8942: Permission denied (13)
qml: Loaded 14 UA override(s) from file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/ua-overrides-mobile.js
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebContext.qml:104: Error: Invalid write to global property "userAgentOverrides"
qml: user agent: Mozilla/5.0 (Linux; Ubuntu 16.04 like Android 9) AppleWebKit/537.36 Chrome/77.0.3865.129 Mobile Safari/537.36
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/index.html:252) caches: 
[PERFORMANCE]: Last frame took 40 ms to render.
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/index.html:240) Uncaught (in promise) TypeError: Failed to register a ServiceWorker: The URL protocol of the current origin ('file://') is not supported.
file:///usr/share/morph-browser/ContentPickerDialog.qml:118: ReferenceError: acceptTypes is not defined
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/PopupBase.qml:166: TypeError: Cannot read property 'theme' of null
[PERFORMANCE]: Last frame took 70 ms to render.
QObject::startTimer: Timers cannot be started from another thread
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/Dialog.qml:239:13: QML Column: Cannot specify top, bottom, verticalCenter, fill or centerIn anchors for items inside Column. Column will not function.
qml: 'caller' is DEPRECATED. It has no effect.
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/Dialog.qml:239:13: QML Column: Cannot specify top, bottom, verticalCenter, fill or centerIn anchors for items inside Column. Column will not function.
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/script.js:105) doBook [object ArrayBuffer] [object Object]
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/script.js:326) metadata [object Object]
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/script.js:299) navigation [object Object]
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/script.js:266) bookKey epubjs:0.3:http://www.gutenberg.org/15859
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/script.js:271) storedLocations ["epubcfi(/6/4[item6]!/4,/2/1:0,/30/1:1)...
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/script.js:565) storedPos epubcfi(/6/4[item6]!/4/2/1:0)
[PERFORMANCE]: Last frame took 32 ms to render.
QObject::killTimer: Timers cannot be stopped from another thread
QObject::startTimer: Timers cannot be started from another thread
```

Log file (`~/.cache/upstart/application-click-simplereader.collaproductions_simplereader_1.0.2.log`) of an affected device (Xperia X):

```
library "libgui.so" wasn't loaded and RTLD_NOLOAD prevented it
Creating a QMirClientScreen now
Started as a local application container.
qml: Loading webview on /home/phablet/.local/share/simplereader.collaproductions
Sandboxing disabled by user.
[18436:18464:1122/223955.154053:ERROR:file_path_watcher_linux.cc(71)] Failed to read /proc/sys/fs/inotify/max_user_watches
qq= WebViewImplOxide_QMLTYPE_108(0x80fea30) 0x8243d90
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onContextMenuIntercepted"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onInsertionHandleTapped"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onStatusChanged"
qml: Start browsing
QDir::mkpath: Empty or null file name
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:145:9: Unable to assign [undefined] to QObject*
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.581\" (uid=32011 pid=18436 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.Operations\" member=\"list\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.581\" (uid=32011 pid=18436 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"running\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.581\" (uid=32011 pid=18436 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"last_error\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.581\" (uid=32011 pid=18436 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"result\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
[18436:18475:1122/223955.759753:ERROR:zygote_host_impl_linux.cc(259)] Failed to adjust OOM score of renderer with pid 18476: Permission denied (13)
qml: Loaded 14 UA override(s) from file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/ua-overrides-mobile.js
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebContext.qml:104: Error: Invalid write to global property "userAgentOverrides"
qml: user agent: Mozilla/5.0 (Linux; Ubuntu 16.04 like Android 9) AppleWebKit/537.36 Chrome/77.0.3865.129 Mobile Safari/537.36
[PERFORMANCE]: Last frame took 66 ms to render.
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/index.html:252) caches: 
[PERFORMANCE]: Last frame took 61 ms to render.
[PERFORMANCE]: Last frame took 49 ms to render.
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/index.html:240) Uncaught (in promise) TypeError: Failed to register a ServiceWorker: The URL protocol of the current origin ('file://') is not supported.
file:///usr/share/morph-browser/ContentPickerDialog.qml:118: ReferenceError: acceptTypes is not defined
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/PopupBase.qml:166: TypeError: Cannot read property 'theme' of null
[PERFORMANCE]: Last frame took 219 ms to render.
QObject::startTimer: Timers cannot be started from another thread
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/Dialog.qml:239:13: QML Column: Cannot specify top, bottom, verticalCenter, fill or centerIn anchors for items inside Column. Column will not function.
qml: 'caller' is DEPRECATED. It has no effect.
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/Dialog.qml:239:13: QML Column: Cannot specify top, bottom, verticalCenter, fill or centerIn anchors for items inside Column. Column will not function.
qt.qpa.mirclient: Attempted to deliver an event to a non-existent window, ignoring.
library "libgui.so" wasn't loaded and RTLD_NOLOAD prevented it
Creating a QMirClientScreen now
Started as a local application container.
qml: Loading webview on /home/phablet/.local/share/simplereader.collaproductions
Sandboxing disabled by user.
[19923:19954:1122/225057.055983:ERROR:file_path_watcher_linux.cc(71)] Failed to read /proc/sys/fs/inotify/max_user_watches
qq= WebViewImplOxide_QMLTYPE_108(0x307e4a40) 0x309293e0
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onContextMenuIntercepted"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onInsertionHandleTapped"
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:144:5: QML Connections: Cannot assign to non-existent property "onStatusChanged"
qml: Start browsing
QDir::mkpath: Empty or null file name
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebView.qml:145:9: Unable to assign [undefined] to QObject*
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.614\" (uid=32011 pid=19923 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.Operations\" member=\"list\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.614\" (uid=32011 pid=19923 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"running\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.614\" (uid=32011 pid=19923 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"last_error\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
error calling result "An AppArmor policy prevents this sender from sending this message to this recipient; type=\"method_call\", sender=\":1.614\" (uid=32011 pid=19923 comm=\"webapp-container --app-id=simplereader.collaproduc\") interface=\"com.canonical.libertine.Service.OperationsMonitor\" member=\"result\" error name=\"(unset)\" requested_reply=\"0\" destination=\"com.canonical.libertine.Service\" (uid=32011 pid=4785 comm=\"/usr/bin/python3 /usr/bin/libertined \")"
lastError - no arguments?
[19923:19960:1122/225057.249100:ERROR:zygote_host_impl_linux.cc(259)] Failed to adjust OOM score of renderer with pid 19962: Permission denied (13)
qml: Loaded 14 UA override(s) from file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/ua-overrides-mobile.js
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Morph/Web/MorphWebContext.qml:104: Error: Invalid write to global property "userAgentOverrides"
qml: user agent: Mozilla/5.0 (Linux; Ubuntu 16.04 like Android 9) AppleWebKit/537.36 Chrome/77.0.3865.129 Mobile Safari/537.36
[PERFORMANCE]: Last frame took 65 ms to render.
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/index.html:252) caches: 
[PERFORMANCE]: Last frame took 32 ms to render.
[PERFORMANCE]: Last frame took 37 ms to render.
qml: [JS] (file:///opt/click.ubuntu.com/simplereader.collaproductions/1.0.2/index.html:240) Uncaught (in promise) TypeError: Failed to register a ServiceWorker: The URL protocol of the current origin ('file://') is not supported.
[PERFORMANCE]: Last frame took 34 ms to render.
[PERFORMANCE]: Last frame took 33 ms to render.
file:///usr/share/morph-browser/ContentPickerDialog.qml:118: ReferenceError: acceptTypes is not defined
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/PopupBase.qml:166: TypeError: Cannot read property 'theme' of null
[PERFORMANCE]: Last frame took 289 ms to render.
QObject::startTimer: Timers cannot be started from another thread
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/Dialog.qml:239:13: QML Column: Cannot specify top, bottom, verticalCenter, fill or centerIn anchors for items inside Column. Column will not function.
qml: 'caller' is DEPRECATED. It has no effect.
file:///usr/lib/aarch64-linux-gnu/qt5/qml/Ubuntu/Components/Popups/1.3/Dialog.qml:239:13: QML Column: Cannot specify top, bottom, verticalCenter, fill or centerIn anchors for items inside Column. Column will not function.
QObject::killTimer: Timers cannot be stopped from another thread
QObject::startTimer: Timers cannot be started from another thread
```

**Comment by nicolascolla** [comment]
@cibersheep, I ask since you commented in https://github.com/nicolascolla/simple-reader/issues/1, is the app working for you now? Or is it displaying this bug, too?

**Comment by cibersheep** [comment]
> @cibersheep, I ask since you commented in #1, is the app working for you now? Or is it displaying this bug, too?

~~It was funny, trying to open an epub from the "open book" button didn't work but it did when tapping on the import arrow icon. If you need it, I can test again and send you logs if you need.~~

Feel free to ping me on Telegram

**Comment by cibersheep** [comment]
Update. This is how I did it:

* case 1
  - Start app
  - Tap on eather the botton or the import icon
  - Select a file `ecdlm-cc.epub`

-> ePub is not loaded

* case 2
  - Start app
  - Tap on button or import icon
  - Cancel Popup
  - Tap on button or import icon again
  - Select a file `ecdlm-cc.epub`

-> ePub loads

The log is from test case 1 with button, case 1 with icon, case 2 with button, case 2 with icon :)

[application-click-simplereader.collaproductions_simplereader_1.0.2.log](https://github.com/nicolascolla/simple-reader/files/7587529/application-click-simplereader.collaproductions_simplereader_1.0.2.log)

**Comment by matv1** [comment]
Some weird observations:

I tested the M10 as well. LionelB says it fails for him but on mine it does not fail. We talked directly about what could be the difference. In the end we matched everything: My device is also the FHD version, I made sure I tested with the exact same file as Lionel used, I also tested on RC channel. In no circumstance did loading this file in Simple Reader fail for me.

I have no idea why this is.
